xtos/include/sof/lib/perf_cnt.h

```diff
@@ -115,12 +134,16 @@ struct perf_cnt_data {
 			(uint32_t)sof_cycle_get_64();				\
 		uint32_t cpu_ts =						\
 			(uint32_t)perf_cnt_get_cpu_ts();			\
-		if ((pcd)->plat_ts) {						\
-			(pcd)->plat_delta_last = plat_ts - (pcd)->plat_ts;	\
-			(pcd)->cpu_delta_last = cpu_ts - (pcd)->cpu_ts;		\
-		}								\
-		(pcd)->plat_ts = plat_ts;					\
-		(pcd)->cpu_ts = cpu_ts;						\
+		if (plat_ts > (pcd)->plat_ts)                    \
+			(pcd)->plat_delta_last = (plat_ts - (pcd)->plat_ts);	\
+		else                                             \
+			(pcd)->plat_delta_last = (UINT32_MAX + plat_ts   \
+									- (pcd)->plat_ts); \
```

Comment:
is this 64-bit arithmetics? If it's 32-bit then that will overflow: `UINT32_MAX + plat_ts` will be calculated first and it doesn't fit in 32 bits. Just change that to `UINT32_MAX - (pcd)->plat_ts) + plat_ts`. And I'm wondering why one would decide to use parentheses in
```
x = (a + b);
```
...

Reply:
Thanks, changed based on your comments, please review again, if no other issues, please +1 for merge.